### PR TITLE
feat: externalize ngWalker to support custom lint rules

### PR DESCRIPTION
**Pull request by piyukore06**
relates to #657

### CODE REVIEW

**Comment by mgechev** [comment]
Thanks for the quick PR! My idea was to externalize the content of the `angular` directory into a separate package. Does this cover your use case completely? Keep in mind that there are more visitor abstractions for traversal of template, style, and expression ASTs.

**Comment by piyukore06** [comment]
Hi @mgechev, Thank you for your input. I only needed NgWalker for my use case, but I think it makes sense to export the angular directory contents, there's so much goodness, everyone could use to build rules.
Just one question, When you are saying a separate package, do you mean npm module like @codelyzer/angular and @codelyzer/rules something like that?

**Comment by piyukore06** [comment]
Yea I am trying but couldn't find anything.. locally the command runs without any problems.

File: src/angular/index.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export * from './templates/basicTemplateAstVisitor';
+export * from './templates/jitReflector';
```
Comment (mgechev) [inline src/angular/index.ts]: I don't think we should export the `jitReflector` as part of the public API.

File: src/angular/index.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+export * from './templates/recursiveAngularExpressionVisitor';
+export * from './templates/referenceCollectorVisitor';
```
Comment (mgechev) [inline src/angular/index.ts]: Better not export `referenceCollectorVisitor`.

File: src/angular/index.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+export * from './templates/templateParser';
+
+export * from './styles/basicCssAstVisitor';
```
Comment (mgechev) [inline src/angular/index.ts]: Can we keep only the `basicCssAstVisitor` exported?

File: src/angular/index.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+export * from './styles/cssAst';
+export * from './styles/cssLexer';
+export * from './styles/cssParser';
+export * from './styles/parseCss';
+export * from './styles/parseUtil';
+
+export * from './config';
+export * from './expressionTypes';
+export * from './metadata';
+export * from './metadataReader';
```
Comment (mgechev) [inline src/angular/index.ts]: Can we skip the `metadataReader`?

File: src/angular/index.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+export * from './ngWalker';
+export * from './ngWalkerFactoryUtils';
+export * from './sourceMappingVisitor';
```
Comment (mgechev) [inline src/angular/index.ts]: My guess is that the error comes from the missing newline.

**Comment by mgechev** [comment]
@piyukore06 if TypeScript won't complain, I'd suggest exporting only the base visitors without any extra internal APIs. This way, we'd be able to make changes in them without introducing issues for consumers of codelyzer.

**Comment by piyukore06** [comment]
@mgechev thank you for your comments! They lead me to a better understanding of, where to look and what should be exported.

**Pull request merged**